helm-chart/binderhub/templates/secret.yaml

```diff
@@ -3,18 +3,17 @@ apiVersion: v1
 metadata:
   name: binder-secret
 type: Opaque
-data:
-  {{- $values :=  dict "config" dict }}
-  {{- $cfg := .Values.config }}
-  binder.hub-token: {{ .Values.jupyterhub.hub.services.binder.apiToken | b64enc | quote }}
-  {{- /* every 'pick' here should be matched with a corresponding 'omit' in secret.yaml */ -}}
-  {{- if $cfg.GitHubRepoProvider }}
-  {{- $_ := set $values.config "GitHubRepoProvider" (pick $cfg.GitHubRepoProvider "client_id" "client_secret" "access_token") }}
-  {{- end }}
-  {{- if $cfg.GitLabRepoProvider }}
-  {{- $_ := set $values.config "GitLabRepoProvider" (pick $cfg.GitLabRepoProvider "access_token" "private_token") }}
-  {{- end }}
-  values.yaml: {{ $values | toYaml | b64enc | quote }}
+stringData:
+  {{- /* Make it easy to mount certain parts without traversing YAML */}}
+  binder.hub-token: {{ .Values.jupyterhub.hub.services.binder.apiToken | quote }}
+
+  {{- /* Stash away all Helm template values */}}
+  values.yaml: |
+    {{- .Values | toYaml | nindent 4 }}
+
+  {{- /* Glob files to allow them to be mounted by the binderhub pod */}}
+  {{- /* key=filename: value=content */}}
+  {{- (.Files.Glob "files/*").AsConfig | nindent 2 }}
```

Comment:
data -> stringData to not need to base64 encode everything, which makes everything easier to read etc.

Reply:
Do you know why it used to be b64 encoded?

Reply:
[ConfigMap resources](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#configmap-v1-core) have `data` and `binaryData`, while [Secret resources](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#secret-v1-core) has `data` and `stringData`.

- stringData expects plain strings and will write out plain strings when mounted.
- binaryData expects base64 encoded strings and will base64 decode those strings before writing out the data to a file on mount

The `data` field on ConfigMap defaults to be a stringData kind of field, while the `data` field on a Secret defaults to be a binaryData kind of field. I'm not sure why they have different defaults though. Base64 doesn't add security in any way though.